pkg/issuer/acme/dns/dns.go

```diff
@@ -239,6 +241,18 @@ func (s *Solver) solverForIssuerProvider(providerName string) (solver, error) {
 			providerConfig.AzureDNS.ResourceGroupName,
 			providerConfig.AzureDNS.HostedZoneName,
 		)
+	case providerConfig.LiveDNS != nil:
+		apiKeySecret, err := s.secretLister.Secrets(s.resourceNamespace).Get(providerConfig.LiveDNS.APIKey.Name)
+		if err != nil {
+			return nil, fmt.Errorf("error getting livedns service account: %s", err)
+		}
+
+		apiKey := strings.TrimSpace(string(apiKeySecret.Data[providerConfig.LiveDNS.APIKey.Key]))
```

Comment:
If a user does not specify a `key` on the secret key selector, should we have a default? If not, we should probably log a more useful error (right now, I guess it'd just print 403 errors due to the api key being set to "")

Reply:
isn't this picked up by validation?

if I omit the `key` field, this condition gets set:

    Resource validation failed: spec.acme.dns01.providers[1].livedns.apiKeySecretRef.key: Required value: secret key is required